compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9065,7 +9073,18 @@ private STNode parseResourceAccessSegmentRhs(boolean isRhsExpr, boolean isInMatc
                 return parseResourceAccessSegmentRhs(isRhsExpr, isInMatchGuard);
         }
     }
-    
+    private STNode parseOnFailCheck() {
+        startContext(ParserRuleContext.CHECK_ON_FAIL);
+        STNode onKeyword = parseOnKeyword();
+        STNode failKeyword = parseFailKeyword();
+        STNode varName = parseVariableName();
+        STNode rightDoubleArrow = parseDoubleRightArrow();
+        STNode errConstructor = parseErrorConstructorExpr(false);
+        endContext();
+        return STNodeFactory.createOnFailCheckNode(null, onKeyword,
```

Comment:
We can't parse `null` for the kind. node.kind is called most of the time. 
Need to add `"kind": "ON_FAIL_CHECK"` property in the `syntax_tree_descriptor.json`
Also, an entry in `SyntaxKind.java`